samples/snippets/src/test/java/pubsublite/spark/SamplesIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+    expected.put("the", 24);
+    expected.put("of", 16);
+    expected.put("and", 14);
+    expected.put("i", 13);
+    expected.put("my", 10);
+    expected.put("a", 6);
+    expected.put("in", 5);
+    expected.put("that", 5);
+    expected.put("soul", 4);
+    expected.put("with", 4);
+    expected.put("as", 3);
+    expected.put("feel", 3);
+    expected.put("like", 3);
+    expected.put("me", 3);
+    expected.put("so", 3);
+    expected.put("then", 3);
+    expected.put("us", 3);
+    expected.put("when", 3);
+    expected.put("which", 3);
+    expected.put("am", 2);
+    Map<String, Integer> actual = new HashMap<>();
+    Queue<PubsubMessage> results =
+        subscriberExample(
+            cloudRegion.value(),
+            cloudZone.zoneId(),
+            projectNumber.value(),
+            destinationSubscriptionName.value());
+    for (PubsubMessage m : results) {
+      String[] pair = m.getData().toStringUtf8().split("_");
+      actual.put(pair[0], Integer.parseInt(pair[1]));
+    }
+    assertThat(actual).containsAtLeastEntriesIn(expected);
+  }
+
+  private void verifySimpleWriteResultViaPSL() {
+    Queue<PubsubMessage> results =
+        subscriberExample(
+            cloudRegion.value(),
+            cloudZone.zoneId(),
+            projectNumber.value(),
+            destinationSubscriptionName.value());
+    // The streaming query runs for 60s, and rate source generate one row per sec.
+    assertThat(results.size()).isGreaterThan(10);
+    for (PubsubMessage m : results) {
+      assertThat(m.getOrderingKey()).isEqualTo("testkey");
+      assertThat(m.getData().toStringUtf8()).startsWith("data_");
+    }
+  }
+
+  private void verifyConsoleOutput(Job job) {
+    Storage storage =
+        StorageOptions.newBuilder().setProjectId(projectId.value()).build().getService();
+    Matcher matches = Pattern.compile("gs://(.*?)/(.*)").matcher(job.getDriverOutputResourceUri());
+    assertThat(matches.matches()).isTrue();
+
+    Blob blob = storage.get(matches.group(1), String.format("%s.000000000", matches.group(2)));
+    String sparkJobOutput = new String(blob.getContent());
+    log.atInfo().log(sparkJobOutput);
+    String expectedWordCountResult =
+        "-------------------------------------------\n"
+            + "Batch: 0\n"
+            + "-------------------------------------------\n"
+            + "+--------------------+---------+------+---+--------------------+"
+            + "--------------------+---------------+----------+\n"
+            + "|        subscription|partition|offset|key|                data|"
+            + "   publish_timestamp|event_timestamp|attributes|\n"
+            + "+--------------------+---------+------+---+--------------------+"
+            + "--------------------+---------------+----------+\n"
+            + "|projects/java-doc...|        0|     0| []|          [61 5F 31]|"
+            + "2021-02-01 23:26:...|           null|        []|\n"
```

Comment:
These have dates hardcoded - are we sure this test will stay valid?

Reply:
These message were published to the Lite topic some time back. New Lite subscriptions by default seek to the beginning of the Lite topic. So these should stay valid.

Reply:
Yep, as Tianzi said, the input messages are fixed.